python_modules/dagster/dagster/_core/definitions/asset_check_spec.py

```diff
@@ -98,6 +98,13 @@ def __new__(
             additional_deps, AssetCheckKey(asset_key, name)
         )
 
+        for dep in additional_asset_deps:
+            if dep.asset_key == asset_key:
+                raise ValueError(
+                    f"Asset check {name} for asset {asset_key.to_string()} cannot have an addtional dependency "
+                    f"on asset{asset_key.to_string()}."
```

Comment:
Can we split out dagster core changes into a separate PR? Also, small typo here:

```suggestion
                    f"Asset check {name} for asset {asset_key.to_string()} cannot have an additional dependency "
                    f"on asset {asset_key.to_string()}."
```

Reply:
https://github.com/dagster-io/dagster/pull/19358